server/dataone_upload.py

```diff
@@ -316,6 +327,51 @@ def create_upload_tale_yaml(tale, remote_objects, item_ids, user, client):
     return pid, len(yaml_file)
 
 
+def upload_license_file(client, license_id):
+    """
+    Upload a license file to DataONE.
+
+    :param client: The client that interfaces DataONE
+    :param license_id: The ID of the license (see `ExtraFileNames` in constants)
+    :type client: MemberNodeClient_2_0
+    :type license_id: str
+    :return: The pid and size of the license file
+    """
+
+    # Holds the license text
+    license_text = str()
+
+    # Path to the license file
+    license_path = os.path.join(ROOT_DIR,
+                                'girder',
+                                'plugins',
+                                'wholetale',
+                                'licenses',
+                                license_files[license_id])
+    try:
```

Comment:
This looks pretty good. What is the behavior of this try/catch block and method as a whole when the call to `f.read()` fails (for any reason) or if `os.path.getsize()` fails?

Reply:
Good catch. I moved `getsize` inside, and return `None, 0` if it fails. I then filter `None` out of the object pid list, and added a check to make sure the license size is greater than 0 when adding the EML record (of the license object).

Here's an example of a package that got a `None, 0` return. The license file is gone, but we were still able to capture the intellectual rights in the EML.
https://dev.nceas.ucsb.edu/#view/089335a5-d8a8-4f6d-8835-d23fa150f7fe

Reply:
Addressed in c82dcf1543937f84cfff193445eb4c4925e699a5